### PR TITLE
feat: 870 task model editor tweaks save changes back into tds

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
Allow users to save models that they have edited.

https://user-images.githubusercontent.com/17088680/226617639-9af1246b-68e4-4b16-8e6c-f5c22bb85efd.mov

Resolves #(issue)
#870

### CODE REVIEW

**Comment by Tom-Szendrey** [comment]
Just looking into what 404 error I am getting and checking if thats because of me or if its because my local has issues

File: packages/client/hmi-client/src/services/provenance.ts [hunk 1]
```diff
@@ -98,12 +98,12 @@ async function getRelatedArtifacts(id: string, rootType: ProvenanceType): Promis
...
-				modelRevisionIDs.length < MAX_RELATED_ARTIFACT_COUNT
-			) {
-				modelRevisionIDs.push(node.id.toString());
-			}
+			// if (
+			// 	node.type === ProvenanceType.ModelRevision &&
+			// 	modelRevisionIDs.length < MAX_RELATED_ARTIFACT_COUNT
+			// ) {
+			// 	modelRevisionIDs.push(node.id.toString());
+			// }
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/provenance.ts:106]: Let's remove commented code form PR.
That's why we use a versioning system.

File: packages/client/hmi-client/tests/unit/petrinet/petrinet.spec.ts [hunk 2]
```diff
@@ -1,268 +1,31 @@
...
 	]
 };
+
 describe('fetchGraphResult', () => {
 	it('Should provide correct graph', () => {
-		const out = parsePetriNet2IGraph(SIRD);
-		assert.equal(out.edges.length, 7); // Technicall 8, one is a multi-edge
-		assert.equal(out.nodes.length, 7);
+		const out = parsePetriNet2IGraph(SIR);
+		assert.equal(out.edges.length, 5); // Technicall 8, one is a multi-edge
```
Comment (YohannParis) [inline packages/client/hmi-client/tests/unit/petrinet/petrinet.spec.ts:22]: I think this comment is irrelevant now

**Review by YohannParis — APPROVED** [review]
Looks great

File: packages/client/hmi-client/src/services/provenance.ts [hunk 3]
```diff
@@ -98,12 +98,6 @@ async function getRelatedArtifacts(id: string, rootType: ProvenanceType): Promis
 				// but the fetch service expects IDs as string(s)
 				datasetIDs.push(node.id.toString());
 			}
-			if (
-				node.type === ProvenanceType.ModelRevision &&
-				modelRevisionIDs.length < MAX_RELATED_ARTIFACT_COUNT
-			) {
-				modelRevisionIDs.push(node.id.toString());
-			}
```
[the remark this answers is not in the document]
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/services/provenance.ts:106]: ^ Will do

**Pull request merged**